frontend/src/components/activePage/TableInstance/RowInstanceHeader/RowInstanceHeader.tsx

```diff
@@ -0,0 +1,219 @@
+import { FC, useState } from 'react';
+import { Typography } from 'antd';
+import { WorkspaceRole } from '../../../../utils';
+import {
+  CaretDownOutlined,
+  CaretUpOutlined,
+  SortAscendingOutlined,
+  SortDescendingOutlined,
+} from '@ant-design/icons';
+
+const { Text } = Typography;
+export interface IRowInstanceHeaderProps {
+  viewMode: WorkspaceRole;
+  showGuiIcon?: boolean;
+  templatePrettyName: string;
+  handleSorting: (sortingType: string, sorting: number) => void;
+  handleManagerSorting: (
+    sortingType: string,
+    sorting: number,
+    sortingTemplate: string
+  ) => void;
+}
+
+const RowInstanceHeader: FC<IRowInstanceHeaderProps> = ({ ...props }) => {
+  const {
+    viewMode,
+    handleSorting,
+    handleManagerSorting,
+    templatePrettyName: tpn,
+  } = props;
+  const [prettyNameOrder, setPrettyNameOrder] = useState(0);
+  const [templatePrettyNameOrder, setTemplatePrettyNameOrder] = useState(0);
+  const [timeStampOrder, setTimeStampOrder] = useState(0);
+  const [tenantDisplayNameOrder, setTenantDisplayNameOrder] = useState(0);
+  const [tenantIdOrder, setTenantIdOrder] = useState(0);
+
+  type sortKey =
+    | 'tenantId'
+    | 'tenantDisplayName'
+    | 'templatePrettyName'
+    | 'timeStamp'
+    | 'prettyName';
+
+  const getSorting = (value: number) => {
+    return value === (0 || -1) ? 1 : -1;
+  };
```

Comment:
```suggestion
  const getSorting = (value: number) => value === (0 || -1) ? 1 : -1;
```
Better to add a line which says what this function does